sdk/src/main/java/io/opentelemetry/sdk/trace/SpanBuilderSdk.java

```diff
@@ -31,30 +33,50 @@
 import io.opentelemetry.trace.TraceId;
 import io.opentelemetry.trace.TraceOptions;
 import io.opentelemetry.trace.Tracestate;
+import io.opentelemetry.trace.unsafe.ContextUtils;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.Random;
 import javax.annotation.Nullable;
 
 /** {@link SpanBuilderSdk} is SDK implementation of {@link Span.Builder}. */
-@SuppressWarnings("unused") // TODO: finish implementation
+@SuppressWarnings("unused")
 class SpanBuilderSdk implements Span.Builder {
+  private static final long INVALID_ID = 0;
+
   private final String spanName;
   private final SpanProcessor spanProcessor;
   private final TraceConfig traceConfig;
+  private final Resource resource;
+
+  private final Clock clock;
+  private final Random random;
```

Comment:
Add a TODO that this can be changed with ThreadLocal implementation. We can discuss later how to do that.